### PR TITLE
Add REST contact api [PD-2514]

### PR DESCRIPTION
**Pull request by tdphillips**

### CODE REVIEW

File: mypartners/views.py [hunk 1]
```diff
@@ -2180,3 +2181,89 @@ def api_create_partner(request):
...
+    Returns a list of contacts. If a parameter named "q" is provided,
+    we filter for contacts whose name contain that value. If a parameter named
+    "partner_id" is provided, we only search that partner's contacts. The
+    contacts whose name start with "q", if any exist, are moved to the
+    beginning of the returned list.
+    """
+    company = get_company_or_404(request)
+
+    q = request.GET.get('q') or request.POST.get('q')
+    partner_id = (request.GET.get('partner_id')
```
Comment (tdphillips) [inline mypartners/views.py]: Names are open for discussion. This can be modified to fit whatever the UI is going to be sending.

**Comment by koonsies** [comment]
Failed test :(

File: mypartners/tests/test_api.py [hunk 2]
```diff
@@ -545,3 +543,105 @@ def test_create_partner(self):
...
+                           email='contact%s@example.com' % i)
+        contact = Contact.objects.last()
+        contact.name = "New Contact"
+        contact.email = "new_contact@example.com"
+        contact.save()
+
+        response = self.client.post(reverse('api_get_contacts'),
+                                    {'q': 'contact'})
+        payload = json.loads(response.content)
+        print payload
```
Comment (koonsies) [inline mypartners/tests/test_api.py]: print statement

**Comment by koonsies** [comment]
:+1: Once @darrint gets a look at this, I think it's ready for merge.

**Pull request merged**